amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -743,6 +754,36 @@ private static String getStringOrEmpty(Object o) {
          return retVal;
     }
      
-    
+     /**
+      * This class is used for parallel processing.
+      *
+      * @author Wessi:kiyako1991@gmail.com
+      */
+
+     static class RunnableTask implements Runnable {
```

Comment:
Using a runnable task you make each thread to work outside the current HttpRequest, thus you loose control of the connections to the database. I ran the code and is started getting `Caused by: org.postgresql.util.PSQLException: FATAL: sorry, too many clients already
    at ` errors and it only process 23 clients since the code is swallowing  the exceptions

Reply:
I faced the same exception and fix it by modifying the pg config file by increasing the size of maximum number of connection to the db.